### ISSUE TITLE
When converting to NSZ using 22_1 thread it justs freezes the progress

### ISSUE THREAD

**Issue by arctco** [julesontheroad/NSC_BUILDER#83]
Basically I have tried to convert suing 22_1 thread (I think that's the name of the option) and it just freezes, the progress bar starts going and then stops, working normally if I choose 22_no threads but the speed is 1-4mbs while with threads it's 50mbs+
What can I do to fix or is it normal?

**Comment by julesontheroad** [comment]
That's normal on certain files it's due to how the zstandard library handles threads. If you check the output file you'll see how it's increasing on size, the library seems to precompress with each thread to memory and then join the result to disk. I don't exactly know in which situations this happens or if it's due to number of memory available but while some files show a constant progress others stop while they dump the compressed data to disk.
Basically the program itself is progressing but the reads on the file stop while the current progress gets it's final compression
